src/js/game/hud/parts/building_placer_logic.js

```diff
@@ -114,6 +120,38 @@ export class HUDBuildingPlacerLogic extends BaseHUDPart {
         this.root.camera.upPostHandler.add(this.onMouseUp, this);
     }
 
+    /**
+     * Returns the current base rotation for the current meta-building.
+     * @returns {number}
+     */
+    get currentBaseRotation() {
+        if (!this.root.app.settings.getAllSettings().rotationByBuilding) {
+            return this.currentBaseRotationGeneral;
+        }
+        const building = this.currentMetaBuilding.get();
+        if (building != undefined && this.preferredBaseRotations.hasOwnProperty(building.getId())) {
+            return this.preferredBaseRotations[building.getId()];
+        } else {
+            return this.currentBaseRotationGeneral;
+        }
+    }
+
+    /**
+     * Sets the base rotation for the current meta-building.
+     */
```

Comment:
Could you add a jsdoc `@param {number} rotation` here? (Sorry for nitpicking :P)

Reply:
Added. No worries, consistent code style is quite helpful for new contributors trying to find their way around.